src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1046,7 +1046,8 @@ private static bool IsDefaultValue(BoundExpression expr)
                 case BoundKind.Conversion:
                     {
                         var conversion = (BoundConversion)expr;
-                        return conversion.Conversion.Kind == ConversionKind.DefaultOrNullLiteral &&
+                        var conversionKind = conversion.Conversion.Kind;
+                        return (conversionKind == ConversionKind.DefaultLiteral || conversionKind == ConversionKind.NullLiteral) &&
                             IsDefaultValue(conversion.Operand);
                     }
                 case BoundKind.DefaultExpression:
```

Comment:
>case BoundKind.DefaultExpression [](start = 16, length = 32)

Should we also handle DefaultLiteral here?

Reply:
Possibly, but looking at the callers of `IsDefaultValue`, I can't come up with a scenario where this would affect nullable diagnostics. 
Default literals should have been converted to DefaultExpression or produced an error in initial binding.
I'll add anyways.

---
In reply to: [315430170](https://github.com/dotnet/roslyn/pull/37596#discussion_r315430170) [](ancestors = 315430170)